### PR TITLE
Feature/mish2/build fix for cuda

### LINKED ISSUE #485: NVCC std::tuple GPU warning
a more human friendly translation of this warning:

`std::tuple`'s operator= is not marked with `__device__`, so it can't be called from a `__device__` context. 

This is confusing, because nvcc is supposed to support `std::tuple` since 9.2:
https://docs.nvidia.com/cuda/archive/9.2/cuda-toolkit-release-notes/index.html#cuda-compiler-new-features

Screen Shot 2021-06-07 at 4 03 14 PM

My guess is that nvcc only supports the `constexpr` parts of `std::tuple` in device code, and since this copy constructor isn't `constexpr` until C++20, it won't work in device code. That being said, if we want to pursue implementing a small `serac::tuple` object, nothing would prevent us from making those constructors `constexpr` and working on GPUs.

_Originally posted by @samuelpmishLLNL in https://github.com/LLNL/serac/pull/478#discussion_r646999352_

### PR DESCRIPTION
**Pull request by samuelpmishLLNL**
These changes were the result of a heroic effort by Chris, Josh and Gregory helping me get TPLs and serac building + linking with cuda on WSL2.

### CODE REVIEW

File: scripts/spack/packages/axom/package.py [hunk 1]
```diff
@@ -256,6 +256,27 @@ def initconfig_hardware_entries(self):
...
         ))
 
+        # Grab lib directory for the current fortran compiler
+        libdir = pjoin(os.path.dirname(
+                       os.path.dirname(self.compiler.fc)),
+                       "lib")
+        description = ("Adds a missing rpath for libraries "
+                       "associated with the fortran compiler")
+
+        if "+cuda" in spec:
```
Comment (white238) [inline scripts/spack/packages/axom/package.py:269]: You dropped a serac edit coment here

**Comment by white238** [comment]
There is also a bunch of extra stuff in this branch.

**Comment by white238** [comment]
Merged Axom changes back into Axom/Spack:

https://github.com/LLNL/axom/pull/593
https://github.com/spack/spack/pull/24349

**Comment by samuelpmishLLNL** [comment]
closed in favor of https://github.com/LLNL/serac/pull/495

**Pull request closed without merging**